Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -242,18 +266,22 @@ class ScriptMessageClientGtk final : public WebScriptMessageHandler::Client {
 
     bool supportsAsyncReply() override
     {
-        return false;
+        return m_supportsAsyncReply;
     }
-    
-    void didPostMessageWithAsyncReply(WebPageProxy&, FrameInfoData&&, API::ContentWorld&, WebCore::SerializedScriptValue&, WTF::Function<void(API::SerializedScriptValue*, const String&)>&&) override
+
+    void didPostMessageWithAsyncReply(WebPageProxy&, FrameInfoData&&, API::ContentWorld&, WebCore::SerializedScriptValue& serializedScriptValue, WTF::Function<void(API::SerializedScriptValue*, const String&)>&& completionHandler) override
     {
+        WebKitUserScriptMessage* message = webKitUserScriptMessageCreate(serializedScriptValue, WTFMove(completionHandler));
+        g_signal_emit(m_manager, signals[SCRIPT_REQUEST_RECEIVED], m_handlerName, message);
```

Comment:
When signals have a return type, we need to pass one last parameter to `g_signal_emit()` to indicate where to put the returned value. In this case we don't need to do anything with it (we are only interested in the effect of the `g_signal_accumulator_true_handled` accumulator), but it still needs to be passed:

```suggestion
        gboolean resultValue;
        g_signal_emit(m_manager, signals[SCRIPT_REQUEST_RECEIVED], m_handlerName, message, &resultValue);
```